tests/integration/test_charm_metrics.py

```diff
@@ -337,3 +368,88 @@ async def test_charm_issues_metrics_for_failed_runner(
         github_repository=forked_github_repository,
         post_job_status=PostJobStatus.REPO_POLICY_CHECK_FAILURE,
     )
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_charm_issues_metrics_for_abnormal_termination(
+    model: Model,
+    app: Application,
+    forked_github_repository: Repository,
+    forked_github_branch: Branch,
+):
+    """
+    arrange: A properly integrated charm with a runner registered on the fork repo.
+    act: Dispatch a test workflow and afterwards kill run.sh. After that, reconcile.
+    assert: The RunnerStart, RunnerStop and Reconciliation metric is logged.
+        The Reconciliation metric has the post job status set to Abnormal.
+    """
+    await app.set_config({"path": forked_github_repository.full_name})
+    await ensure_charm_has_runner(app=app, model=model)
+
+    unit = app.units[0]
+
+    workflow = forked_github_repository.get_workflow(
+        id_or_file_name=DISPATCH_CRASH_TEST_WORKFLOW_FILENAME
+    )
+    assert workflow.create_dispatch(forked_github_branch, {"runner": app.name})
+
+    # Wait for the workflow log to be picked up by the runner
+    sleep(60)
+
+    # Make the runner terminate abnormally by killing run.sh
```

Comment:
I see, perhaps clarify in the workflow file that we do something to cause a failure and need to time to get it done